apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ast/Ast.kt

```diff
@@ -27,37 +27,32 @@ internal data class CodeGenerationAst(
       val operationId: String,
       val queryDocument: String,
       val variables: List<InputField>,
-      val dataType: OperationDataType,
-      val filePath: String
+      val dataType: ObjectType,
   ) {
     enum class Type {
       QUERY, MUTATION, SUBSCRIPTION
     }
   }
 
-  data class OperationDataType(
-      val rootType: TypeRef,
-      val nestedTypes: ObjectTypeContainer
-  )
-
   data class FragmentType(
+      val name: String,
       val graphqlName: String,
-      val rootType: TypeRef,
-      val defaultImplementation: TypeRef,
-      val nestedTypes: ObjectTypeContainer,
+      val description: String,
+      val interfaceType: ObjectType,
+      val implementationType: ObjectType,
       val fragmentDefinition: String
   )
 
   data class ObjectType(
       val name: String,
       val description: String,
-      val deprecated: Boolean,
-      val deprecationReason: String,
+      val deprecationReason: String?,
       val fields: List<Field>,
+      val nestedObjects: List<ObjectType>,
       val implements: Set<TypeRef>,
-      val schemaType: String?,
       val kind: Kind,
-      val typeRef: TypeRef
+      val typeRef: TypeRef,
+      val introspectionSchemaType: IntrospectionSchema.TypeRef?,
```

Comment:
It looks like this is only used in `ObjectType.primaryConstructorSpec`, can we use `introspectionSchemaType.rawType.name` instead ? That'll make the Ast more independant of the introspection.